### DISCUSSION TITLE
Docker build doesn't work via GitHub Actions, but it works locally

### DISCUSSION THREAD

**Discussion by Ruitjes** [community/community#25215] [Actions]
<p>Hello everyone!</p>
<p>I’ve a weird issue with docker build in combination with GitHub actions. I’m totally new with Docker files and I really, really, really appreciate some help.</p>
<p>When running the command on my CLI, it works perfectly fine. But when trying it on GitHub action, I’m receiving the error:</p>
<p><code>unable to prepare context: path "..." not found Error: Process completed with exit code 1.</code></p>
<p>Command:<br>
<code>cd services/Gateway/OcelotGateway</code><br>
<code>docker build -f Dockerfile ... -t stanpanman/ocelotgateway:1.0.0</code></p>
<p>My GitHub Action</p>
<pre><code class="lang-auto">  build:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v2
      - name: docker login
        env:
          DOCKER_USER: '${{ secrets.DOCKER_USER }}'
          DOCKER_PASSWORD: '${{ secrets.DOCKER_PASSWORD }}'
        run: |
          docker login -u $DOCKER_USER -p $DOCKER_PASSWORD
      - name: docker build ocelotgateway
        run: |
          cd services/Gateway/OcelotGateway 
          docker build ... -t stanpanman/ocelotgateway:1.0.0
</code></pre>
<p>My Docker image:</p>
<pre><code class="lang-auto">FROM mcr.microsoft.com/dotnet/aspnet:6.0 AS base
WORKDIR /app
EXPOSE 80
EXPOSE 443

FROM mcr.microsoft.com/dotnet/sdk:6.0 AS build
WORKDIR /src
COPY ["Gateway/OcelotGateway/OcelotGateway.csproj", "Gateway/OcelotGateway/"]
RUN dotnet restore "Gateway/OcelotGateway/OcelotGateway.csproj"
COPY . .
WORKDIR "/src/Gateway/OcelotGateway"
RUN dotnet build "OcelotGateway.csproj" -c Release -o /app/build

FROM build AS publish
RUN dotnet publish "OcelotGateway.csproj" -c Release -o /app/publish

FROM base AS final
WORKDIR /app
COPY --from=publish /app/publish .
ENTRYPOINT ["dotnet", "OcelotGateway.dll"]
</code></pre>
<p>My file structure<br>
</p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/4/e/4e89e050d173c3b11eb513d0a04b6ce419ac8257.jpeg" data-download-href="/uploads/short-url/bcMDKCV5y93c9KhRntB8L9xACBV.jpeg?dl=1" title="" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/26769441/181089626-cfa01f7b-a98b-48e4-81a9-d166706130a6.jpeg" alt="" data-base62-sha1="bcMDKCV5y93c9KhRntB8L9xACBV" width="690" height="216" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/4/e/4e89e050d173c3b11eb513d0a04b6ce419ac8257_2_690x216.jpeg, https://aws1.discourse-cdn.com/github/optimized/3X/4/e/4e89e050d173c3b11eb513d0a04b6ce419ac8257_2_1035x324.jpeg 1.5x, https://aws1.discourse-cdn.com/github/optimized/3X/4/e/4e89e050d173c3b11eb513d0a04b6ce419ac8257_2_1380x432.jpeg 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/4/e/4e89e050d173c3b11eb513d0a04b6ce419ac8257_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use href="#far-image"></use></svg><span class="filename"></span><span class="informations">2104×660 110 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by airtower-luna** [comment, marked as answer]
<aside class="quote no-group" data-username="Ruitjes" data-post="1" data-topic="244236">
<div class="title">
<div class="quote-controls"></div>
 Ruitjes:</div>
<blockquote>
<p><code>docker build -f Dockerfile ... -t stanpanman/ocelotgateway:1.0.0</code></p>
</blockquote>
</aside>
<p>The one argument you give to <code>docker build</code> without an option first is the build context: The directory in which Docker will look for the Dockerfile, and the content of which it’ll use while building the image. In your case: <code>...</code></p>
<p>In general <code>...</code> isn’t a valid directory. There are two very similar things that usually are: <code>.</code> (the current directory, very commonly used as build context), and <code>..</code> (the parent directory). Replace the three dots with the directory you actually want. :smiley_cat:</p>
<aside class="quote no-group" data-username="Ruitjes" data-post="1" data-topic="244236">
<div class="title">
<div class="quote-controls"></div>
 Ruitjes:</div>
<blockquote>
<p>My Docker image:</p>
</blockquote>
</aside>
<p>As a side note, what you’re quoting there is a <em>Dockerfile</em>. The <em>image</em> is the result of the build.</p>

**Comment by Ruitjes** [comment]
<p>Yey!</p>
<p>The issue is solved by using the command like this :grin::</p>
<pre><code class="lang-auto">          cd services
          docker build -f Ticket/Ticket.API/Dockerfile . -t stanpanman/ticketapi:1.0.0
</code></pre>
<p>Still not sure what the difference is between my previous command.</p>

**Comment by Ruitjes** [comment]
<p><a class="mention" href="/airtower-luna">@airtower-luna</a> Thanks for your response, and your corrections!<br>
My suspicion was also that there was something wrong with the “…”. The strange thing is that it works fine when I try it on my MacBook, but when I run the same command in GitHub Actions, it can’t be found.</p>
<p>I replaced the three dots with what I think is the good directory, it’s able to find the Dockerfile but now I end up with this error:</p>
<pre><code class="lang-auto">        run: |
          docker build -f services/Gateway/OcelotGateway/Dockerfile . -t stanpanman/ocelotgateway:1.0.0
</code></pre>
<p>Error:</p>
<pre><code class="lang-auto">Step 7/17 : COPY ["Gateway/OcelotGateway/OcelotGateway.csproj", "Gateway/OcelotGateway/"]
COPY failed: file not found in build context or excluded by .dockerignore: stat Gateway/OcelotGateway/OcelotGateway.csproj: file does not exist
</code></pre>

**Comment by airtower-luna** [comment]
<p>Do those files exist <em>relative to the build context</em>? I notice the same directory names in the path to the Dockerfile, but with another parent directory.</p>
<p>You can change the context in the build command or the paths in the Dockerfile, but they must fit together.</p>
